plugins/modules/zos_copy.py

```diff
@@ -0,0 +1,1513 @@
+            {str} -- Destination where the file was copied to
+        """
+        src = self.module.params.get("src")
+        dest = self.module.params.get("dest")
+        if src_ds_type in MVS_SEQ.union(MVS_PARTITIONED):
+            self._mvs_copy_to_uss(src, dest, src_ds_type, src_member, member_name=member_name)
+        else:
+            if os.path.isfile(temp_path or conv_path or src):
+                dest = self._copy_to_file(src, dest, conv_path, temp_path)
+            else:
+                dest = self._copy_to_dir(src, dest, conv_path, temp_path)
+
+        if self.common_file_args is not None:
+            mode = self.common_file_args.get("mode")
+            group = self.common_file_args.get("group")
+            owner = self.common_file_args.get("owner")
+            if mode is not None:
+                self.module.set_mode_if_different(dest, mode, False)
+            if group is not None:
+                self.module.set_group_if_different(dest, group, False)
+            if owner is not None:
+                self.module.set_owner_if_different(dest, owner, False)
+        return dest
+
+    def _copy_to_file(self, src, dest, conv_path, temp_path):
+        """Helper function to copy a USS src to USS dest.
+
+        Arguments:
+            src {str} -- USS source file path
+            dest {str} -- USS dest file path
+            temp_path {str} -- Path to the location where the control node transferred data to
+            conv_path {str} -- Path to the converted source file or directory
+
+        Returns:
+            {str} -- Destination where the file was copied to
+        """
+        if os.path.isdir(dest):
+            dest = os.path.join(dest, os.path.basename(src) if src else 'inline_copy')
+
+        new_src = temp_path or conv_path or src
+        try:
+            if self.is_binary:
+                copy.copy_uss2uss_binary(new_src, dest)
+            else:
+                shutil.copy(new_src, dest)
+        except OSError as err:
+            self.fail_json(
+                msg="Destination {0} is not writable".format(dest),
+                stderr=str(err)
+            )
+        except Exception as err:
+            self.fail_json(
+                msg="Unable to copy file {0} to {1}".format(new_src, dest),
+                stderr=str(err)
+            )
+        return dest
+
+    def _copy_to_dir(self, src_dir, dest_dir, conv_path, temp_path):
+        """Helper function to copy a USS directory to another USS directory
+
+        Arguments:
+            src_dir {str} -- USS source directory
+            dest {str} -- USS dest directory
+            temp_path {str} -- Path to the location where the control node transferred data to
+            conv_path {str} -- Path to the converted source directory
+
+        Returns:
+            {str} -- Destination where the directory was copied to
+        """
+        new_src_dir = temp_path or conv_path or src_dir
+        if os.path.exists(dest_dir):
```

Comment:
@ddimatos , yes, backup is taken before performing the rmtree/copytree operation. However, the backup path is not returned to the user on module failure. The backup path is inly returned on module success. Perhaps it makes sense that it should be returned regardless of failure and success, as long as `backup` or `backup_file` is provided.

Reply:
Its worth considering, maybe it goes against the norm but it would be nice to know that if something went really wrong i could restore my data, wondering if @blakeinate has seen such behavior in other modules.

Reply:
I think returning the backup path makes sense in any case where a backup is created. 

I may have to go back and make sure *zos_encode* follows the same behavior.